### ISSUE TITLE
Ouroboros' unusual introspection habits

### ISSUE THREAD

**Issue by Tallefer** [dom96/snake#8]
A quick clockwise U-turn was expected, but
![nim-snake](https://user-images.githubusercontent.com/395476/30326907-c9422aa8-97d2-11e7-8b87-ad76ab5af0f8.png)
This is not supposed to happen, right?
There is also a possibility that it is a side effect of a temporary slowdown or/and disconnection.

**Comment by dom96** [comment]
Yeah, this shouldn't happen. I'm guessing it's not easy to reproduce though :\

**Comment by Tallefer** [comment]
But the main question is - what to blame? Is it Nim compiler or this game code (less likely IMO) or is it browser's VM async quirks? What do you think? Is it even important at all to raise this question?

**Comment by dom96** [comment]
It's most definitely some race condition. I don't think it happens very often though, does it?

**Comment by Tallefer** [comment]
Yup, same conditions are not *too* common. Well, I can only try to reproduce it at some point, but as I said - probably related to disconnection and freeze which were occurring at that very moment, Memory was almost full, too. So browser in general was chugging a lot then.

**Comment by dom96** [comment]
This _might_ be fixed now :)

**Comment by soypat** [comment]
I managed to reproduce python introspection by making a fast counter-clockwise turn (uparrow->leftarrow) while transiting the top row. Version: https://nim-lang.org/features.html

**Comment by matomatical** [comment]
^ The same issue affects me. I think this affects the topmost and leftmost rows and I could reproduce it reliably a few times. I already opened a new issue describing the problem, sorry for the double-up: #11.

(Although it does seem like a distinct issue, because (1) it's reproducible (2) it affects the whole leftmost and topmost rows (3) rather than requiring fast turns it seems there is a whole time step in which the head is missing---all of this suggests some kind of off-by-one-error in the game logic for wraparound at the left/top side, rather than a race condition)
